### ISSUE TITLE
bionic: Installing flatpaks hangs on `F: Calling system helper: Deploy` and finally aborts with org.freedesktop.DBus.Error.NoReply

### ISSUE THREAD

**Issue by bentolor** [flatpak/flatpak#2014]
## Linux distribution and version
Ubuntu 18.04 bionic, upgraded from 16.04 

## Flatpak version
`Flatpak 1.0.0` via PPA

## Description of the problem
I tried with the distribution version as well the ppa version to:

```
flatpak install --ostree-verbose -vv --from https://flathub.org/repo/appstream/de.haeckerfelix.gradio.flatpakref
```
this causes a download an then hangs long time on `F: Calling system helper: Deploy` and finally aborts with `GDBus.Error:org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying`. 

Here are the last lines of the terminal output

```
[###################-] Downloading files: 21460/21463 297,3 MB (343,0 kB/s)F: write of 1fc02f6f9e576473ff405424433016564e6015c3388c1e2b750266d62e70de08.file complete
F: write of 403d3f920705e75c2abd3293037f8d00f1453eb46ba0b8ca519f764230be3b80.file complete
F: write of 4daff22c08d709c8d8a90a2fb508c64c5359ad3e14745469355063d7a0cf2c8c.file complete
F: pull: idle, exiting mainloop
F: Committing transaction in repository 0x55b5180c8bc0
F: txn commit staging-a4eab392-ad93-4d83-a48f-2a1bf2bdd9b6-NFeD8p
F: Popping lock non-blocking with timeout 30
F: Pop lock: state=shared, depth=1
F: Unlocking repo
[####################] 976 metadata, 20487 content objects fetched; 290375 KiB transferred in 867 secondsF: Calling system helper: Deploy
F: Free lock: state=unlocked, depth=0
F: Closing repo lock file
F: Free lock table

Fehler: Failed to installieren org.gnome.Platform/x86_64/3.28: GDBus.Error:org.freedesktop.DBus.Error.NoReply: Message recipient disconnected from message bus without replying
```

## Steps to reproduce
Reproducible on my machine, not sure if happens on others.

**Comment by alexlarsson** [comment]
Should be fixed by https://github.com/flatpak/flatpak/pull/2010

**Comment by alexlarsson** [comment]
Or rather, *may* be fixed by it. Can you test?

**Comment by bentolor** [comment]
Thanks @alexlarsson for having a look at my issue. 

Is there a nightly PPA/prebuilt `.deb` available or can I find somewhere some hints to build a `.deb` on my own?

**Comment by alexlarsson** [comment]
Hmm, actually, that bug was about DeployAppstream, whereas this is Deploy. I wonder if it is something similar though.

**Comment by alexlarsson** [comment]
Between the last outout and the NoReply error, did it take about 30 seconds?

**Comment by bentolor** [comment]
I would assume much longer as I aborted in most time the process and in the case shown above I left the terminal unattended and returned only minutes later.

**Comment by alexlarsson** [comment]
If you can reproduce this, can you, before running the install, start a root shell and run `/usr/lib/flatpak/flatpak-system-helper -v -r --no-idle-exit`. Then paste the output from this during the install.

**Comment by bentolor** [comment]
Ok – will try. But I won't have physical access to the machine until Friday and yet I failed to create a xrdp-session. So I need some time.

Thanks again for your assitance.

**Comment by bentolor** [comment]
Hi @alexlarsson – I was able to test again on the machine. Since my last try I received a new update v1.0.1 via PPA The behavior has now changed a little.

**TL;DR:**
* **Installation with v1.0.1 takes a very looong pause (5~10 Minutes), but now eventually continues**
* **I can't find `gradio` in my desktop menu nor command line. But running via `flatpak run` now works**

On the client-side the deploy hangs for a very long time (~5 minutes !) at the Line `Calling system helper: Deploy`.  Different to my previous runs, now it finally progresses. But I'm not sure if Gradio is correctly installed.

### Client-side output
```
ben@neptun ~  $ flatpak install --ostree-verbose -vv --from https://flathub.org/repo/appstream/de.haeckerfelix.gradio.flatpakref

...

F: pull: idle, exiting mainloop
F: Committing transaction in repository 0x55b590640620
F: txn commit staging-a4eab392-ad93-4d83-a48f-2a1bf2bdd9b6-28pfrk
F: Popping lock non-blocking with timeout 30
F: Pop lock: state=shared, depth=1
F: Unlocking repo
[####################] 21 metadata, 97 content objects fetched; 21607 KiB transferred in 10 secondsF: Calling system helper: Deploy
F: Free lock: state=unlocked, depth=0
F: Closing repo lock file
F: Free lock table

Jetzt bei 009f6ddd2144.
Installieren: de.haeckerfelix.gradio/x86_64/stable von flathub
F: Preparing transaction in repository 0x55b590640bc0
F: Pushing lock non-blocking with timeout 30
F: Creating repo lock table
F: Opening repo lock file
F: Push lock: state=unlocked, depth=0
F: Locking repo shared
F: Using new tmpdir staging-a4eab392-ad93-4d83-a48f-2a1bf2bdd9b6-dMNiik
F: Adding GPG keyring data from flathub.trustedkeys.gpg to verifier
F: starting fetch of 39a2ee5142c37a05277cae9d69779371e1e74937a638c4b85ffaf32cc4027a22.commit (detached)
F: fetch of 39a2ee5142c37a05277cae9d69779371e1e74937a638c4b85ffaf32cc4027a22.commit (detached) complete
F: starting fetch of 39a2ee5142c37a05277cae9d69779371e1e74937a638c4b85ffaf32cc4027a22.commit
F: _ostree_fetcher_should_retry_request: error: unset, n_retries_remaining: 5
F: fetch of 39a2ee5142c37a05277cae9d69779371e1e74937a638c4b85ffaf32cc4027a22.commit complete
F: Adding GPG keyring data from flathub.trustedkeys.gpg to verifier
F: _ostree_fetcher_should_retry_request: error: unset, n_retries_remaining: 5
F: write of 39a2ee5142c37a05277cae9d69779371e1e74937a638c4b85ffaf32cc4027a22.commit complete
F: pull: idle, exiting mainloop
F: Adding GPG keyring data from flathub.trustedkeys.gpg to verifier
F: starting fetch of 39a2ee5142c37a05277cae9d69779371e1e74937a638c4b85ffaf32cc4027a22.commit (detached)
F: fetch of 39a2ee5142c37a05277cae9d69779371e1e74937a638c4b85ffaf32cc4027a22.commit (detached) complete
F: _ostree_fetcher_should_retry_request: error: unset, n_retries_remaining: 5
F: Adding GPG keyring data from flathub.trustedkeys.gpg to verifier
F: starting fetch of 7d93ded960a91c03ae46695073ec1a1fa8187b2b25c8bc90751051895d9f6ccc.dirtree
F: fetch of 7d93ded960a91c03ae46695073ec1a1fa8187b2b25c8bc90751051895d9f6ccc.dirtree complete
F: _ostree_fetcher_should_retry_request: error: unset, n_retries_remaining: 5

...

F: write of 13928f1a8bbe4adb3740faa7f8e62d11ea7eb3f47ebfd0606235903703498249.file complete
F: pull: idle, exiting mainloop
F: Committing transaction in repository 0x55b5906409e0
F: txn commit staging-a4eab392-ad93-4d83-a48f-2a1bf2bdd9b6-OvwD5S
F: Popping lock non-blocking with timeout 30
F: Pop lock: state=shared, depth=1
F: Unlocking repo
[####################] 7 metadata, 2 content objects fetched; 5 KiB transferred in 0 secondsF: Calling system helper: Deploy
F: Free lock: state=unlocked, depth=0
F: Closing repo lock file
F: Free lock table

Jetzt bei 19dc2550698a.
F: Calling system helper: RunTriggers
```

### Output of `flatpak-system-helper`

```
 ben@neptun ~  $ flatpak --version
Flatpak 1.0.1
 ben@neptun ~  $ sudo su -
Willkommen zu fish, der freundlichen interaktiven Shell
root@neptun ~# /usr/lib/flatpak/flatpak-system-helper -v -r --no-idle-exit
F: Bus acquired, creating skeleton
F: Name acquired
F: Deploy /var/tmp/flatpak-cache-L52EOZ/repo-cSaO7U 0 runtime/org.gnome.Platform/x86_64/3.28 flathub 
      [ loooooooong delay here ]
F: No checksum specified, getting tip of runtime/org.gnome.Platform/x86_64/3.28 from origin flathub
F: tip resolved to: 9d3fe926d425ad837e40a0945020a00f09c1f531335987a753e07f276d76ee36
F: Deploy /var/tmp/flatpak-cache-L52EOZ/repo-5sGITn 0 runtime/org.freedesktop.Platform.GL.nvidia-390-48/x86_64/1.4 flathub 
F: No checksum specified, getting tip of runtime/org.freedesktop.Platform.GL.nvidia-390-48/x86_64/1.4 from origin flathub
F: tip resolved to: d3ee7d2d7c896877152aa9d8f58b4e3e69ce84786fffc630664f8502e48f2e1c
F: extracting extra data to /var/lib/flatpak/runtime/org.freedesktop.Platform.GL.nvidia-390-48/x86_64/1.4/.d3ee7d2d7c896877152aa9d8f58b4e3e69ce84786fffc630664f8502e48f2e1c-XJTMOZ/files/extra
F: Disallowing ipc access
F: Disallowing network access
F: Failed to run in transient scope: No systemd user session available, cgroups not available
F: Running /app/bin/apply_extra 
F: Deploy /var/tmp/flatpak-cache-L52EOZ/repo-gI6d5b 0 runtime/org.freedesktop.Platform.ffmpeg/x86_64/1.6 flathub 
F: No checksum specified, getting tip of runtime/org.freedesktop.Platform.ffmpeg/x86_64/1.6 from origin flathub
F: tip resolved to: d757f762489e444551d17223590635d4f4014a36b6fecd73323bf3f95f1aa490
F: Deploy /var/tmp/flatpak-cache-L52EOZ/repo-YOvaD5 0 runtime/org.gtk.Gtk3theme.Ambiance/x86_64/3.22 flathub 
F: No checksum specified, getting tip of runtime/org.gtk.Gtk3theme.Ambiance/x86_64/3.22 from origin flathub
F: tip resolved to: 73fed99df212c179f776452d1eb4f49d9e8fd87ba6f8434816db2e9c6c92ea19
F: Deploy /var/tmp/flatpak-cache-L52EOZ/repo-W9hoQV 0 runtime/org.gnome.Platform.Locale/x86_64/3.28 flathub 
F: No checksum specified, getting tip of runtime/org.gnome.Platform.Locale/x86_64/3.28 from origin flathub
F: tip resolved to: 009f6ddd2144edac7e5299b6997859bd91f200032418af8aaf026e0760d70410
F: Deploy /var/tmp/flatpak-cache-L52EOZ/repo-6cuezH 0 app/de.haeckerfelix.gradio/x86_64/stable flathub 
F: No checksum specified, getting tip of app/de.haeckerfelix.gradio/x86_64/stable from origin flathub
F: tip resolved to: 39a2ee5142c37a05277cae9d69779371e1e74937a638c4b85ffaf32cc4027a22
F: Deploy /var/tmp/flatpak-cache-L52EOZ/repo-bjwoG5 0 runtime/de.haeckerfelix.gradio.Locale/x86_64/stable flathub 
F: No checksum specified, getting tip of runtime/de.haeckerfelix.gradio.Locale/x86_64/stable from origin flathub
F: tip resolved to: 19dc2550698a49e443bed7eb82d1e8f531aa566f39546a074020590e260f2a60
F: RunTriggers 
F: running triggers from /usr/share/flatpak/triggers
F: running trigger desktop-database.trigger
F: running trigger gtk-icon-cache.trigger
F: running trigger mime-database.trigger
```

### Current situation

Flatpak lists gradio as installed, but I'm unable to find it via Menu nor command-line

```
 ▲  ben@neptun ~  $ flatpak list
Referenz                                             Optionen      
de.haeckerfelix.gradio/x86_64/stable                 system,current
org.freedesktop.Platform.GL.nvidia-390-48/x86_64/1.4 system,runtime
org.freedesktop.Platform.ffmpeg/x86_64/1.6           system,runtime
org.gnome.Platform/x86_64/3.28                       system,runtime
org.gtk.Gtk3theme.Ambiance/x86_64/3.22               system,runtime
 ben@neptun ~  $ gradio

Der Befehl 'gradio' wurde nicht gefunden, meinten Sie:

```

I would expect this to be a local install, but `~/.local/share/flatpak/` contains only a `repo` subdir, but I was able to find a `/var/lib/flatpak/app/de.haeckerfelix.gradio/`.

Running via `flatpak run de.haeckerfelix.gradio/x86_64/stable` seems to work. 

Now my only problem is, that I'm unsure if it is correct, that 

1. I fail to find/access Gradio via the Gnome 3 shell and the command-line directly.
2. The installation procedure takes a very long pause similar to my previous runs

**Comment by bentolor** [comment]
After a restart of Gnome3 the icon appears and everything seems now to work as expected except the very long hang during install.

So If you don't see anything wrong here, this ticket can be closed. 

Thank you very much for your support.

**Comment by claviola** [comment]
@alexlarsson 
I see a similar issue, but I'm not sure if it's the same thing. I can't properly install flatpak on Ubuntu 19.04, either the default version from the repos or the newer one from the PPA. The install also takes several minutes and ends up with a DBus error: `Error org.freedesktop.DBus.Error.NoReply: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.`

I *was* able to download and run a flatpak packaged app, though, so I'm not sure how broken my setup is now. How can I check?
